### PR TITLE
[specific ci=Group6-VIC-Machine] Resource suggestion fixes for vic-machine

### LINKED ISSUE #4203: 6-10-List failure on VC
Ran the command:
bin/vic-machine-linux ls --target xxxx.xxx.vmware.com/ --thumbprint=xxx --user xxx --password=xxx --compute-resource cls

```
Mar 8 2017 18:39:26.874-06:00 INFO ### Listing VCHs ####
 Mar 8 2017 18:39:27.043-06:00 INFO Validating target 
Mar 8 2017 18:39:27.043-06:00 INFO Validating compute resource
 Mar 8 2017 18:39:27.055-06:00 INFO Suggesting valid values for --compute-resource based on "cls" 
Mar 8 2017 18:39:27.057-06:00 INFO Failed to find resource pool in the provided path, showing all top level resource pools.
 Mar 8 2017 18:39:27.060-06:00 INFO Suggested values for --compute-resource: 
Mar 8 2017 18:39:27.060-06:00 INFO "cls"
 Mar 8 2017 18:39:27.060-06:00 ERROR -------------------- 
Mar 8 2017 18:39:27.060-06:00 ERROR resource pool 'cls' not found 
Mar 8 2017 18:39:27.060-06:00 ERROR List cannot continue - compute resource validation failed: validation of configuration failed
 Mar 8 2017 18:39:27.060-06:00 ERROR --------------------
 Mar 8 2017 18:39:27.060-06:00 ERROR vic-machine-linux ls failed: list failed
```

Weird error, it suggests cls, but at the same time indicates cls is not valid, only difference being single quotes vs double quotes?

### PR DESCRIPTION
**Pull request by dougm**
- Support vic-machine create targeting Datacenter within an inventory folder.

- Simplify compute-resource suggestions.
  Removes quite a bit of code, which also did not work when clusters/hosts were within inventory folders.
  Some of the removed code was working around the old govmomi Finder "list mode" limitations.

- Add vcsim based tests for vic-machine validation against multiple resources of all types,
  within inventory folders.  These tests also cover most of the error paths.

- Add LicenseManager support to vcsim

Fixes #4203

### CODE REVIEW

**Comment by dougm** [comment]
tl;dr - When using inventory folders: 1) Fixes suggestions for compute-resource.  2) Fix so vic-machine can target a Datacenter inside a folder.

Diff is on the larger side, but just removes a bunch of code we *don't* need, adds a bunch of test code we *do* need.  

I'll be doing more testing, but please review.

**Comment by jzt** [comment]
So many deletions, it's like music to my eyes. :)
